apps/validate/validate.py

```diff
@@ -435,8 +435,8 @@ def _validate(self, doc, fields=False, **kwargs):
             v.allow_unknown = True
             try:
                 v.validate(doc['validate'], validation_schema)
-            except TypeError as e:
-                logger.exception('Invalid validator schema value "%s" for ' % str(e))
+            except TypeError as ex:
```

Comment:
just curious, why changing it? I'm always using `e` and I see no reason to write one more char here.

Reply:
It's a mypy bug I think. It was complaining about var `e` used outside of except block :)
https://github.com/superdesk/superdesk-core/pull/1965/files#diff-4d8b5e2477ab2c27d07c028d86b5cf71R442

Reply:
Would it be possible to fix Mypy (update?) or ignore this error in its settings? That could save my wrist/fingers for future damages.

Reply:
@jerome-poisson , I agree that this part is dangerous for wrists, will check how to fix/ignore it.

Reply:
@jerome-poisson the issue is that mypy complains about reassigning another type to variable `e` right after try/except block (https://github.com/superdesk/superdesk-core/pull/1965/files#diff-4d8b5e2477ab2c27d07c028d86b5cf71R442). It's a generic hint and I think it's useful to have it switched on. Python does not really care about it, but from code style point of view it's better to have this hint on.
What I don't understand in this particular case is why mypy complains about it "here", because variable `e` is only defined in `except` context.
I will explore it and possibly switch it off if possible, the current behavior is weird IMO.

Reply:
Unfortunately, there is no way to switch it off.
Mypy has only this:

> allow_redefinition (bool, default False)
> Allows variables to be redefined with an arbitrary type, as long as the redefinition is in **the same block and nesting level as the original definition**.

Reply:
isn't the error rather this then? `for e in error_list:`

Reply:
It is, variable e in for e in error_list context is used in more places that e from except TypeError as e: context, so I've changed the one which requires fewer changes :)

Reply:
I see